plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/actions/datasource/DataSourceToolbarUtils.java

```diff
@@ -55,16 +59,18 @@ public static DBPDataSourceContainer getCurrentDataSource(IWorkbenchWindow workb
     public static void refreshSelectorToolbar(IWorkbenchWindow window) {
         if (window instanceof WorkbenchWindow) {
             MTrimBar topTrim = ((WorkbenchWindow) window).getTopTrim();
+            boolean showConnectionSelector = false;
+            IEditorPart activeEditor = window.getActivePage().getActiveEditor();
+            DBPDataSourceContainer dataSourceContainer = null;
+            if (activeEditor instanceof IDataSourceContainerProvider) {
+                showConnectionSelector = true;
+                dataSourceContainer = ((IDataSourceContainerProvider) activeEditor).getDataSourceContainer();
+            }
+            DBPProject resourceProj = EditorUtils.getFileProject(activeEditor.getEditorInput());
+            Boolean canChangeConn = resourceProj == null || resourceProj.hasRealmPermission(RMConstants.PERMISSION_PROJECT_RESOURCE_EDIT);
```

Comment:
Please don't use such abbreviations

```suggestion
            boolean canChangeConn = resourceProj == null || resourceProj.hasRealmPermission(RMConstants.PERMISSION_PROJECT_RESOURCE_EDIT);
```